sdk/cmake/host-tools.cmake

```diff
@@ -15,10 +15,41 @@ function(setup_host_tools)
         set(HOST_EXE_SUFFIX ".exe")
     endif()
 
+    if(CMAKE_SYSTEM_PROCESSOR EQUAL CMAKE_HOST_SYSTEM_PROCESSOR)
+        set(HOST_TOOLS_CMAKE_COMMAND "${CMAKE_COMMAND}")
+        message("Not cross-compiling, no special host-tools cmake command")
+    elseif(MSVC)
+        message("Compiling on ${CMAKE_HOST_SYSTEM_PROCESSOR} for ${CMAKE_SYSTEM_PROCESSOR} (MSVC)")
+        set(HOST_TOOLS_CMAKE_COMMAND "${REACTOS_BINARY_DIR}/host-tools/cmake_shim.cmd")
+        if(MSVC_VERSION EQUAL 1900)
+            file(WRITE ${HOST_TOOLS_CMAKE_COMMAND}
+                "@\"$ENV{VCINSTALLDIR}\\vcvarsall.bat\" /clean_env\n"
+                "@\"$ENV{VCINSTALLDIR}\\vcvarsall.bat\" ${CMAKE_HOST_SYSTEM_PROCESSOR}\n"
+                "\"${CMAKE_COMMAND}\" %*"
+            )
+
+            #set(HOST_TOOLS_CMAKE_COMMAND "$ENV{VCINSTALLDIR}\\vcvarsall.bat" ${CMAKE_HOST_SYSTEM_PROCESSOR} && "${CMAKE_COMMAND}")
+        elseif(MSVC_VERSION GREATER_EQUAL 1910)
+            # 2017 and 2019 use the same folder structure
+            file(WRITE ${HOST_TOOLS_CMAKE_COMMAND}
+                "@\"$ENV{VCINSTALLDIR}\\Auxiliary\\Build\\vcvarsall.bat\" /clean_env\n"
+                "@\"$ENV{VCINSTALLDIR}\\Auxiliary\\Build\\vcvarsall.bat\" ${CMAKE_HOST_SYSTEM_PROCESSOR}\n"
+                "\"${CMAKE_COMMAND}\" %*"
+            )
+           #set(HOST_TOOLS_CMAKE_COMMAND "$ENV{VCINSTALLDIR}\\Auxiliary\\Build\\vcvarsall.bat" ${CMAKE_HOST_SYSTEM_PROCESSOR} && "${CMAKE_COMMAND}")
+        else()
+            message(FATAL "Unable to figure out vcvarsall path")
+        endif()
```

Comment:
If you can make sense of the CMAKE_GENERATOR_TOOLSET stuff & al, this may be used instead. Bu this is fine as it is.

Reply:
good idea, I'll try that first.

Reply:
that does not seem to work, for example ninja does not allow specifying an architecture :(

Reply:
`CMAKE_GENERATOR_*` requires 'msbuild'/VSS: 'ninja' (+ msvc) has no support for these.